docs/AdvancedUsage.md

```diff
@@ -0,0 +1,172 @@
+| **mappingNames**: List<String> | []            | Names of the mapping items included in this recording. Only required for analysis packages unless there is only one recording (not used for packages).                                                                                                                               |
+| **deviceName**: String         | ""            | Optional device name. Required if the recording contains multiple devices and the mapping between device names in the test configuration and device names in the recording cannot be realized by format details. See "Supported file formats" in ecu.test help for more information. |
+| **formatDetails**: String      | ""            | Optional format details. Required if the recording cannot be interpreted with default format details. See "Supported file formats" in ecu.test help for more information.                                                                                                            |
+
+## CheckPackageResult
+
+All given properties can be read via a getter method. (e.g. `.getResult()`)
+
+| Properties                                | Description                                           |
+|-------------------------------------------|-------------------------------------------------------|
+| **result**: String                        | Result of the package check execution                 |
+| **testCasePath**: String                  | Package path of the check package                     |
+| **issues**: List\<HashMap\<String,String> | List of problems found in the package check execution |
+
+## GenerationResult
+
+All given properties can be read via a getter method. (e.g. `.getGenerationResult()`)
+
+| Properties                    | Description                               |
+|-------------------------------|-------------------------------------------|
+| **generationResult**: String  | Result of the report generation execution |
+| **generationMessage**: String | ecu.test report generation message        |
+| **reportOutputDir**: String   | location of ecu.test report               |
+
+## TestResult
+
+All given properties can be read via a getter method. (e.g. `.getReportId()`)
+
+| Properties             | Description                               |
+|------------------------|-------------------------------------------|
+| **reportId**: String   | reportId given by ecu.test                |
+| **testResult**: String | Result of the package / project execution |
+| **reportDir**: String  | location of the report                    |
+
+## UploadResult
+
+All given properties can be read via a getter method. (e.g. `.getUploadResult()`)
+
+| Properties                | Description                                |
+|---------------------------|--------------------------------------------|
+| **uploadResult**: String  | Result of the report upload to test guide. |
```

Comment:
Which upload results are possible or how can the user check if the upload was successful?
Failed uploads has no `reportLink`